eras/conway/impl/src/Cardano/Ledger/Conway/Rules/Epoch.hs

```diff
@@ -256,26 +256,47 @@ epochTransition = do
         , reCurrentEpoch = eNo
         }
     govStateToSeq = Seq.fromList . Map.toList
-    ratSig = RatifySignal . govStateToSeq . unGovActionsState $ cgGovActionsState govSt
-  rs@RatifyState {rsFuture} <-
-    trans @(EraRule "RATIFY" era) $ TRC (ratEnv, cgRatifyState govSt, ratSig)
-  let es'' =
-        epochState'
-          { esAccountState = acnt''
-          , esLState = (esLState epochState') {lsUTxOState = utxoSt'''}
-          }
-          & prevPParamsEpochStateL .~ pp
-          & curPParamsEpochStateL .~ pp
-      -- TODO can we be more efficient?
-      newGov = GovActionsState . Map.fromList . toList $ rsFuture
-      esGovernanceL = esLStateL . lsUTxOStateL . utxosGovStateL
-      esDonationL :: Lens' (EpochState era) Coin
-      esDonationL = esLStateL . lsUTxOStateL . utxosDonationL
-      donations = es'' ^. esDonationL
+    ratSig = RatifySignal . govStateToSeq . prevGovActionsState $ cgGovActionsState govSt
+  RatifyState {rsRemoved, rsEnactState} <-
+    trans @(EraRule "RATIFY" era) $
+      TRC
+        ( ratEnv
+        , RatifyState
+            { rsRemoved = mempty
+            , rsEnactState = govSt ^. cgEnactStateL
+            }
+        , ratSig
+        )
+  let
+    lState = returnProposalDeposits rsRemoved $ esLState epochState'
+    es'' =
+      epochState'
+        { esAccountState = acnt''
+        , esLState = lState {lsUTxOState = utxoSt'''}
+        }
+        & prevPParamsEpochStateL .~ pp
+        & curPParamsEpochStateL .~ pp
+    esGovernanceL :: Lens' (EpochState era) (GovState era)
+    esGovernanceL = esLStateL . lsUTxOStateL . utxosGovStateL
+    oldGovActionsState =
+      es ^. esGovernanceL . cgGovActionsStateL . curGovActionsStateL
+    newGovActionsState =
+      foldr'
+        (\(k, _) m -> Map.delete k m)
+        oldGovActionsState
+        rsRemoved
+    newGov =
+      GovActionsState
+        { curGovActionsState = newGovActionsState
+        , prevGovActionsState = newGovActionsState
```

Comment:
Should this be `prevGovActionsState = oldGovActionsState` instead? I'm not sure and don't have the full picture yet.

Reply:
The idea is that we set both of them to the value we got at the end of the current epoch and only the current state is going to be changed in the next epoch while the previous one stays the same so we can add up the votes.

Reply:
Would you mind adding this as a comment?  Might be useful to remember  in the future, since it does seem a little peculiar if you don't know what's going on.

Reply:
Yes, I 100% agree with Teodora. Adding a comment would be super useful. It took me a while as well to figure out why both of them have to be set to the same value.